azcredentials/cloud.go

```diff
@@ -21,6 +21,8 @@ func GetAzureCloud(settings *azsettings.AzureSettings, credentials AzureCredenti
 		return c.AzureCloud, nil
 	case *AzureClientSecretOboCredentials:
 		return c.ClientSecretCredentials.AzureCloud, nil
+	case *AzureClientPasswordCredentials:
+		return settings.GetDefaultCloud(), nil
```

Comment:
I wonder if we should make this configurable - I'm not sure if this works for other cloud types but we can't discount that it does.